src/Compilers/CSharp/Portable/Binder/Binder_Attributes.cs

```diff
@@ -121,14 +121,14 @@ internal partial class Binder
             BindingDiagnosticBag diagnostics)
         {
             beforeAttributePartBound?.Invoke(node);
-            var boundAttribute = new ExecutableCodeBinder(node, this.ContainingMemberOrLambda, this).BindAttribute(node, boundAttributeType, diagnostics);
+            var boundAttribute = new ExecutableCodeBinder(node, this.ContainingMemberOrLambda, this).BindAttribute(node, boundAttributeType, (this as ContextualAttributeBinder)?.AttributedMember, diagnostics);
```

Comment:
>(this as ContextualAttributeBinder) [](http://example.com/codeflow?start=141&length=35)

Are there valid cases where the `ContextualAttributeBinder` is in the `Next` chain rather than `this`?

Reply:
No, if a ContextualAttributeBinder is present, it will be at the top of the chain here. I did actually consider eliminating the type entirely but I was concerned it could result in too much additional churn.

Reply:
> if a ContextualAttributeBinder is present, it will be at the top of the chain here

Consider asserting that.
```
Debug.Assert(this is ContextualAttributeBinder || !GetAllBinders().Any(this is ContextualAttributeBinder));
```

Reply:
I wasn't able to write a reasonable assertion in a scenario like the following.

https://github.com/dotnet/roslyn/blob/777f4e0a58eda195a6c66a0327267b16bbbf8f9d/src/Compilers/CSharp/Test/Semantic/Semantics/LambdaTests.cs#L3736-L3754

In this case, the BoundAttribute for `[B]` doesn't have errors, so there's not something conveniently available in scope for us to grab onto that lets us skip asserting that "ContextualAttributeBinder is only found on the top".

Reply:
Also, I did try eliminating ContextualAttributeBinder, but I found we still depend on it in `Binder.GetForwardedToAssembly` to break cycles.